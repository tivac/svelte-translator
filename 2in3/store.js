import { Store } from "svelte2/store.js";

class Svelte3Store extends Store {
    subscribe(fn) {
        const { cancel } = this.on("state", ({ current }) => fn(current));

        fn(this.get());

        return cancel;
    }

    update(fn) {
        this.set(fn(this.get()));
    }
}

export default Svelte3Store;

