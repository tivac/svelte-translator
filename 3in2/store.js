import { Store as Svelte2Store } from "svelte2/store.js";

export class Store extends Svelte2Store {
    constructor(store) {
        super();

        this._store = store;
        this._set = this.set;
        
        const unsubscribe = store.subscribe((value) => {
            // Objects are splatted
            if(typeof value === "object" && !Array.isArray(value)) {
                return this.set(value);
            }

            // Everything else is set to the "value" property
            this._set({ value });
        });

        this.unsubscribe = unsubscribe;

        this.set = (args) => {
            if(!this._store.set) {
                throw new Error("Called .set() on a readable store");
            }

            this._set(args);
        };
    }
};
