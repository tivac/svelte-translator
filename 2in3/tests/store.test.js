"use strict";
import { tick } from "svelte";

import { simple, extended } from "./specimens/store.js";

describe("2in3 store adapter", () => {
    const { default : Simple } = require("./specimens/simple.svelte");
    const { default : Extended } = require("./specimens/extended.svelte");
    const { default : Derived } = require("./specimens/derived.svelte");
    const { default : Reactive } = require("./specimens/reactive.svelte");
    
    let root;

    beforeEach(() => {
        root = document.createElement("div");
    });

    afterEach(() => {
        simple.set({
            one : 1,
            two : 2,
        });
        
        extended.set({
            one : 1,
            two : 2,
        });
    });

    it("should render simple store values", async () => {
        new Simple({
            target : root
        });
        
        expect(root.innerHTML).toMatchSnapshot();
        
        simple.set({
            two : 3
        });

        await tick();
        
        expect(root.innerHTML).toMatchSnapshot();
    });
    
    it("should render extended store values", async () => {
        new Extended({
            target : root
        });
        
        expect(root.innerHTML).toMatchSnapshot();
        
        extended.set({
            two : 3
        });

        await tick();
        
        expect(root.innerHTML).toMatchSnapshot();
    });

    it("should render derived store values", async () => {
        new Derived({
            target : root
        });
        
        expect(root.innerHTML).toMatchSnapshot();
        
        simple.set({
            two : 3
        });

        await tick();
        
        expect(root.innerHTML).toMatchSnapshot();
    });
    
    it("should be able to reactively update from store values", async () => {
        new Reactive({
            target : root
        });
        
        expect(root.innerHTML).toMatchSnapshot();
        
        simple.set({
            two : 3
        });

        await tick();
        
        expect(root.innerHTML).toMatchSnapshot();
    });
});
