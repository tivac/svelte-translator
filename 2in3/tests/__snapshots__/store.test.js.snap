// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2in3 store adapter should be able to reactively update from store values 1`] = `
<div>
  1 2 3
</div>
`;

exports[`2in3 store adapter should be able to reactively update from store values 2`] = `
<div>
  1 3 4
</div>
`;

exports[`2in3 store adapter should have a .update() that works 1`] = `
Object {
  "one": 1,
  "three": 3,
  "two": 2,
}
`;

exports[`2in3 store adapter should render derived store values 1`] = `
<div>
  1 2 3
</div>
`;

exports[`2in3 store adapter should render derived store values 2`] = `
<div>
  1 3 4
</div>
`;

exports[`2in3 store adapter should render extended store values 1`] = `
<div>
  1 2 {"one":1,"two":2}
</div>
`;

exports[`2in3 store adapter should render extended store values 2`] = `
<div>
  1 3 {"one":1,"two":2}
</div>
`;

exports[`2in3 store adapter should render simple store values 1`] = `
<div>
  1 2
</div>
`;

exports[`2in3 store adapter should render simple store values 2`] = `
<div>
  1 3
</div>
`;
